### DISCUSSION TITLE
What interface do you recommend for repeatable options?

### DISCUSSION THREAD

**Discussion by Silex** [magit/transient#300] [Support]
Hello,

Say you have some binary that allows to pass environement variables like this:

```
./binary -e foo=123 -e bar=456 -e DEBUG
```

Right now I use the following definition:

``` emacs
   ("-e" "Environment" "-e " read-string)
```

Currently the workaround I found is that if one wants to pass multiple environnement vars, he types the first `-e` through transient, and for the value he inputs `foo=123 -e bar=456`.

Did you face a similar situation? What is typically done?

For example maybe we could make a custom reader with a convention like when you type `foo=123,bar=456`, it would split on the comma and make the multiple `-e` flags.

From my understanding, #194 would not really help because the flag has to be passed each time.

**Comment by tarsius** [comment]
This has already been discussed at https://github.com/Silex/docker.el/issues/169 and linked issues (and it wasn't pleasant).

It has been a while, I know, and I haven't presented another solution still. But I haven't forgotten about this and will at some point look into it again.

**Reply by tarsius** [reply]
Just recording all the relevant issues for when I have time to look into this some more.

https://github.com/Silex/docker.el/issues/137
https://github.com/Silex/docker.el/issues/169
https://github.com/Silex/docker.el/issues/180

**Comment by Silex** [comment]
Thanks. If I was a part of why this was not pleasant for you let me know.

Ok, then I'll close these issues for now, I feel like they are open since too long. I'll ping you if someone does something toward implementing a solution for this use case.

**Reply by tarsius** [reply]
It's all good.
